js/misc/parentalControlsManager.js

```diff
@@ -60,8 +60,24 @@ var ParentalControlsManager = class {
                 logError(e, 'Failed to get parental controls settings');
             }
         }
+
+        this._manager.connect('app-filter-changed', (manager, uid) => {
+            let current_uid = Shell.util_get_uid();
+            // Emit 'changed' signal only if app-filter is changed for currently logged-in user.
+	    if (current_uid == uid)
+                this._manager.get_app_filter_async(current_uid, Malcontent.GetAppFilterFlags.NONE,
+                                                   null, this._onAppFilterChanged.bind(this));
```

Comment:
Any reason to make it async? We already do it synchronously above.

Reply:
Ideally `_async` should be the default go-to approach. Here, we are just relaying the `app-filter-changed` signal (via another `changed` signal) to relevant entities that require an update if the app-filter is changed. Note, we only fire `changed` signal from `ParentalControlsManager` only when this _async call is finished. so, probably it seems fine to me. I didn't notice any problems using `_async` here.

 We do synchronously above because it's quite difficult to make it `_async` work properly. Note, this is the case only for construction time code-paths. Read https://phabricator.endlessm.com/T26969#729762 maybe?

Reply:
I see, yes, the code seems fine and should work (as you said we only fire `changed` when the async call finishes) but I was actually in doubt because of the comment you pointed out in the ticket heh - I misread it as "lets use sync for now everywhere", but I understand now you were referring to async calls on constructors only.